velox/functions/prestosql/StringFunctions.cpp

```diff
@@ -403,6 +403,131 @@ class StringPosition : public exec::VectorFunction {
   }
 };
 
+/**
+ * strpos(string, substring) → bigint
+ * Returns the starting position of the first instance of substring in string.
+ * Positions start with 1. If not found, 0 is returned.
+ *
+ * strpos(string, substring, instance) → bigint
+ * Returns the position of the N-th instance of substring in string. instance
+ * must be a positive number. Positions start with 1. If not found, 0 is
+ * returned.
+ **/
+class StringPositionFromEnd : public exec::VectorFunction {
```

Comment:
Why do we need a vector function? This can likely be written using the simple function API with the same level of performance.

Reply:
@pedroerp , this is a good question. Actually initially I implemented using simple API but then discovered **strpos** implemented in this way so I decided to mimick the behavior. I think if the agreement is that vector function does provide any added-value we can have a follow-up PR with switching all implementations that rely on it.

Reply:
I suspect that the reason is that the strpos function was written before the simple function API was in place. To avoid us from iterating and reviewing the vector code (which is more complicated and might end up not being needed), could we either converge into the simple function version, or write a small micro-benchmark to justify if the vectorized version actually has perf benefits?

Reply:
If we see that the simple function version is just as fast, we could also refactor the strpos function to simplify it.

Reply:
@pedroerp It seems like all functions in that file are implemented as VectorFunctions actually(at first glance hard to spot obvious perf gain). I'd prefer adding strrpos following the same pattern to have a consistent file and have a separate activity of assessing/moving all those to simple function API. As far as the review is concerned the actual implementation in StringCore is going to stay the same as well as the the test cases. We'll have to review only declaration part in the follow-up pr which should be pretty straightforward bulk activity.